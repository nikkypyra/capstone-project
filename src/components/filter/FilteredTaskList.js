import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import TasksStyled from '../tasks/TasksStyled'

FilteredTaskList.propTypes = {
  pets: PropTypes.array.isRequired,
  filteredTasks: PropTypes.array.isRequired,
  handleCheckbox: PropTypes.func.isRequired,
}
export default function FilteredTaskList({
  filteredTasks,
  pets,
  handleCheckbox,
}) {
  return (
    <>
      {filteredTasks
        .slice()
        .sort(
          (taskA, taskB) =>
            taskA.date.localeCompare(taskB.date) ||
            taskA.time.localeCompare(taskB.time)
        )
        .map((todo) => (
          <TaskWrapper key={todo.id}>
            <TasksStyled todo={todo} handleCheckbox={handleCheckbox} />
            <PetName>
              <p>
                For: <strong>{petName(todo).toUpperCase()}</strong>
              </p>
            </PetName>
          </TaskWrapper>
        ))}
    </>
  )
  function petName(todo) {
    const pet = pets.find((pet) => pet.id === todo.petId)
    return pet.name
  }
}

const TaskWrapper = styled.section`
  margin: 16px 0px;
  display: grid;
  grid-template-columns: 1.5fr 2fr 3fr 1fr 1fr;
  grid-template-rows: auto;
  justify-content: space-evenly;
  align-items: center;
`
const PetName = styled.div`
  grid-row: 4/5;
  grid-column: 2/5;
`
