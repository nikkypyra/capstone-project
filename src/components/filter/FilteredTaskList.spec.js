import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import FilteredTaskList from './FilteredTaskList'

afterEach(cleanup)

const petsMock = [{ name: 'Fluffy', id: '1' }]
const filteredTasksMock = [
  {
    id: '1',
    petId: '1',
    description: '',
    date: '',
    time: '',
    person: '',
    complete: true,
  },
]
const handleCheckbox = jest.fn()

describe('<FilteredTaskList />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <FilteredTaskList
          pets={petsMock}
          filteredTasks={filteredTasksMock}
          handleCheckbox={handleCheckbox}
        />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
