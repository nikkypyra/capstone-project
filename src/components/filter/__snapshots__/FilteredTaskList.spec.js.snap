// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilteredTaskList /> should render correctly 1`] = `
<DocumentFragment>
  <section
    class="FilteredTaskList__TaskWrapper-sc-172fr29-0 iQDROY"
  >
    <div
      class="TasksStyled__Marker-j2i96a-0 kTppzS"
    >
      <img
        alt=""
        src="taskpaw.png"
      />
    </div>
    <div
      class="TasksStyled__Description-j2i96a-1 dAkghC"
    >
      <h3 />
    </div>
    <div
      class="TasksStyled__Time-j2i96a-2 DoAZa"
    >
      <h4 />
    </div>
    <div
      class="TasksStyled__Date-j2i96a-3 kHFcCQ"
      data-cy="date"
    >
      <p />
    </div>
    <div
      class="TasksStyled__Person-j2i96a-4 jToTCz"
      data-cy="owner_name"
    >
      <p>
        To be completed by: 
      </p>
    </div>
    <div
      class="TasksStyled__Status-j2i96a-5 fCOCIY"
    >
      <input
        checked=""
        class="Checkbox__CheckboxStyled-sc-1blqffb-0 gefyIW"
        type="checkbox"
      />
    </div>
    <div
      class="FilteredTaskList__PetName-sc-172fr29-1 fIDhRg"
    >
      <p>
        For: 
        <strong>
          FLUFFY
        </strong>
      </p>
    </div>
  </section>
</DocumentFragment>
`;
